braindecode/augmentation/transforms.py

```diff
@@ -0,0 +1,1000 @@
+        to 1Hz.
+    sfreq : float, optional
+        Sampling frequency of the signals to be filtered. Defaults to 100 Hz.
+    max_freq : float | None, optional
+        Maximal admissible frequency. The low cut frequency will be sampled so
+        that the corresponding high cut frequency + transition are below
+        `max_freq`. If omitted or `None`, will default to the Nyquist frequency
+        (`sfreq / 2`).
+    random_state: int | numpy.random.Generator, optional
+        Seed to be used to instantiate numpy random number generator instance.
+        Defaults to None.
+
+    References
+    ----------
+    .. [1] Cheng, J. Y., Goh, H., Dogrusoz, K., Tuzel, O., & Azemi, E. (2020).
+       Subject-aware contrastive learning for biosignals. arXiv preprint
+       arXiv:2007.04871.
+    .. [2] Mohsenvand, M. N., Izadi, M. R., & Maes, P. (2020). Contrastive
+       Representation Learning for Electroencephalogram Classification. In
+       Machine Learning for Health (pp. 238-253). PMLR.
+    .. [3] Park, D.S., Chan, W., Zhang, Y., Chiu, C., Zoph, B., Cubuk, E.D.,
+       Le, Q.V. (2019) SpecAugment: A Simple Data Augmentation Method for
+       Automatic Speech Recognition. Proc. Interspeech 2019, 2613-2617
+    """
+
+    def __init__(
+        self,
+        probability,
+        magnitude=None,
+        mag_range=(0, 2),
+        bandwidth=1,
+        sfreq=100,
+        max_freq=50,
+        random_state=None
+    ):
+        assert isinstance(bandwidth, Real) and bandwidth >= 0,\
+            "bandwidth should be a non-negative float."
+        assert isinstance(sfreq, Real) and sfreq > 0,\
+            "sfreq should be a positive float."
+        assert isinstance(max_freq, Real) and max_freq > 0,\
+            "max_freq should be a positive float."
+        nyq = sfreq / 2
+        if max_freq is None or max_freq > nyq:
+            max_freq = nyq
+        assert bandwidth < max_freq,\
+            f"`bandwidth` needs to be smaller than `max_freq`={max_freq}"
+
+        # override bandwidth value when a magnitude is passed
+        self.sfreq = sfreq
+        self.max_freq = max_freq
+        if magnitude is not None:
+            min_val, max_val = mag_range
+            self.bandwidth = magnitude * max_val + (1 - magnitude) * min_val
+        else:
+            self.bandwidth = bandwidth
+
+        # Handle case of band of length 0
+        if self.bandwidth == 0:
+            operation = identity
+        else:
+            operation = random_bandstop
+
+        super().__init__(
+            probability=probability,
+            operation=operation,
+            magnitude=magnitude,
+            mag_range=mag_range,
+            bandwidth=self.bandwidth,
+            max_freq=self.max_freq,
+            sfreq=self.sfreq,
+            random_state=random_state,
+        )
+
+
+class FrequencyShift(Transform):
+    """Add a random shift in the frequency domain to all channels.
+
```

Comment:
```suggestion

Note that here, the shift is the same for all channels of a single sample.

```

Reply:
Nice precision 💪 I think I will just replace `sample` by `example` here, as the former is a bit ambiguous for signals.